lsp-methods.el

```diff
@@ -785,38 +785,32 @@ interface Range {
   ;;   lsp-on-change:(start,end,length)=(19,19,8)
 
   (if (eq length 0)
-      ;; Adding something only, work from start only
-      `(:range ,(lsp--range (lsp--point-to-position start)
-                            (lsp--point-to-position start))
-               :rangeLength 0
-               :text ,(buffer-substring-no-properties start end))
+    ;; Adding something only, work from start only
+    `(:range ,(lsp--range (lsp--point-to-position start)
+                (lsp--point-to-position start))
+       :rangeLength 0
+       :text ,(buffer-substring-no-properties start end))
 
     (if (eq start end)
-        ;; Deleting something only
-        (if (lsp--bracketed-change-p start end length)
-            ;; (if (and (eq start (plist-get lsp--before-change-vals :start) )
-            ;;          (eq length (- (plist-get lsp--before-change-vals :end)
-            ;;                        (plist-get lsp--before-change-vals :start))))
-            ;; The before-change value is valid, use it
-            `(:range ,(lsp--range (lsp--point-to-position start)
-                                  (plist-get lsp--before-change-vals :end-pos))
-                     :rangeLength ,length
-                     :text "")
-          (lsp--change-for-mismatch start end length))
-      ;; Deleting some things, adding others
+      ;; Deleting something only
       (if (lsp--bracketed-change-p start end length)
-          ;; The before-change value is valid, use it
-          `(:range ,(lsp--range (lsp--point-to-position start)
-                                (plist-get lsp--before-change-vals :end-pos))
-                   :rangeLength ,length
-                   :text ,(buffer-substring-no-properties start end))
-        (lsp--change-for-mismatch start end length)))))
+        ;; The before-change value is bracketed, use it
+        `(:range ,(lsp--range (lsp--point-to-position start)
+                    (plist-get lsp--before-change-vals :end-pos))
+           :rangeLength ,length
+           :text "")
```

Comment:
Are you sure this will always be the empty string? I guess we are in the "deleting only" zone, so should be.